.travis.yml

```diff
@@ -60,6 +77,11 @@ before_install:
 before_script:
   # to stop the script after an error/warning
   - set -e
+  # workaround for OSX traivs issue
+  # /Users/travis/.rvm/scripts/functions/support: line 57: shell_session_update: command not found
+  - if [[ "$TRAVIS_OS_NAME" == "osx" ]]; then unset -f cd; fi
+  - if [[ "$TRAVIS_OS_NAME" == "osx" ]]; then shell_session_update() { :; }; fi
```

Comment:
I am highly entertained by this hack. Nice solution. 

I get why you redefine the buggy `shell_session_update` but what does the  `unset -f cd;` do here?

Reply:
I found the solution here https://github.com/nim-lang/Nim/pull/7962/commits/c9a8ada722a945b6de8c4c145ade25526545a4cc but yeah, it's an interesting workaround.

As far as I understand from http://pubs.opengroup.org/onlinepubs/9699919799/utilities/V3_chap02.html#tag_18_29 `unset -f cd` will unset the function `cd`, which is rather strange.

And yeah, then we create a dummy function for `shell_session_update()`. Let me see if it works without the former.

Reply:
https://github.com/travis-ci/travis-ci/issues/8703#issuecomment-347881274
https://github.com/travis-ci/travis-ci/issues/8703#issuecomment-348439258